modules/openapi-generator/src/main/resources/python/api_response.mustache

```diff
@@ -1,25 +1,73 @@
 """API response object."""
 
 from __future__ import annotations
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Union
+{{#asyncio}}
+from aiohttp import ClientResponse
+{{/asyncio}}
 from pydantic import Field, StrictInt, StrictStr
+{{^asyncio}}
+from urllib3 import HTTPResponse
+{{/asyncio}}
 
 class ApiResponse:
     """
-    API response object
+    API response object returned by `_with_http_info` API methods.
     """
 
     status_code: Optional[StrictInt] = Field(None, description="HTTP status code")
     headers: Optional[Dict[StrictStr, StrictStr]] = Field(None, description="HTTP headers")
     data: Optional[Any] = Field(None, description="Deserialized data given the data type")
-    raw_data: Optional[Any] = Field(None, description="Raw data (HTTP response body)")
+    _raw_data: Union[None, str, bytes] = Field(description="Raw data (HTTP response body)")
+    {{#asyncio}}
+    aiohttp_response: ClientResponse = Field(description="Raw aiohttp response for response streaming")
+    {{/asyncio}}
+    {{^asyncio}}
+    urllib3_response: HTTPResponse = Field(description="Raw urllib3 response for response streaming")
+    {{/asyncio}}
 
-    def __init__(self,
-                 status_code=None,
-                 headers=None,
-                 data=None,
-                 raw_data=None) -> None:
+    def __init__(
+        self,
+        status_code=None,
+        headers=None,
+        data=None,
+        *,
+        _raw_data,
+        {{#asyncio}}
+        aiohttp_response,
+        {{/asyncio}}
+        {{^asyncio}}
+        urllib3_response,
+        {{/asyncio}}
+    ) -> None:
         self.status_code = status_code
         self.headers = headers
         self.data = data
-        self.raw_data = raw_data
+        self._raw_data = _raw_data
+        {{#asyncio}}
+        self.aiohttp_response = aiohttp_response
+        {{/asyncio}}
+        {{^asyncio}}
+        self.urllib3_response = urllib3_response
+        {{/asyncio}}
+
+    @property
+    def raw_data(self) -> Union[str, bytes]:
+        """Get raw response body.
+
+        If `preload_content=False`, this loads the entire response body into
+        memory. If `preload_content=True`, the response body is already loaded
+        earlier.
+        To stream responses, read from `self.{{#asyncio}}aiohttp{{/asyncio}}{{^asyncio}}urllib3{{/asyncio}}_response`.
+
+        :return: `str` if `preload_content=True`, `bytes` if not.
+        """
+        if self._raw_data is None:
+            {{#asyncio}}
+            loop = asyncio.get_event_loop()
+            return loop.run_until_complete(self.aiohttp_response.read())
+            {{/asyncio}}
```

Comment:
You should really not do that here: if a client is using asyncio, I think it will expect to have full control over the async requests. Calling `run_until_complete` here makes the code synchronous which can *really* affect how asynchronous clients work.

Either load the data before during the caller asynchronous method, or transform the property into a method that would be marked `async` for the asyncio client.

Reply:
I have to admit I haven't used asyncio myself so far, so I don't have a clue what I'm doing here :) 

I added this to provide backwards compatibility for users accessing `ApiResponse.raw_data` when using `_preload_content=False`. I just realized that setting `_preload_content=False` is not working for the aiohttp generator on current master, so I guess there is no backwards-compatibility needed here.

I will remove this code.